visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -35,15 +33,15 @@ export class CodeMapArrowService
 		this.scale()
 	}
 
-	onBuildingSelected(selectedBuilding: CodeMapBuilding) {
-		if (this.isEdgeApplicableForBuilding(selectedBuilding)) {
+	onBuildingSelected = (data: { building: CodeMapBuilding }) => {
```

Comment:
Why do you choose to make this type less restrictive?

Reply:
I haven't. It is just wrapped in an object with a key "building" to align with the EventEmitter schema